Src/LinearSolvers/MLMG/AMReX_MLCurlCurl_K.H

```diff
@@ -133,152 +423,179 @@ void mlcurlcurl_adotx_z (int i, int j, int k, Array4<Real> const& Az,
-                        GpuArray<Real,AMREX_SPACEDIM> const& dxinv, int redblack)
+void mlcurlcurl_gs4 (int i, int j, int k,
+                     Array4<Real> const& ex,
+                     Array4<Real> const& ey,
+                     Array4<Real> const& ez,
+                     Array4<Real const> const& rhsx,
+                     Array4<Real const> const& rhsy,
+                     Array4<Real const> const& rhsz,
+#if (AMREX_SPACEDIM == 2)
+                     Real beta,
+#endif
+                     GpuArray<Real,AMREX_SPACEDIM> const& adxinv,
+                     int color, LUSolver<AMREX_SPACEDIM*2,Real> const& lusolver,
+                     CurlCurlDirichletInfo const& dinfo,
+                     CurlCurlSymmetryInfo const& sinfo)
 {
-    constexpr Real omega = Real(1.15);
+    if (dinfo.is_dirichlet_node(i,j,k)) { return; }
 
-    if ((i+j+k+redblack) % 2 != 0) { return; }
+    int my_color = i%2 + 2*(j%2);
+    if (k%2 != 0) {
+        my_color = 3 - my_color;
+    }
 
 #if (AMREX_SPACEDIM == 2)
-    amrex::ignore_unused(ez);
-    Real dyy = dxinv[1] * dxinv[1];
-    Real dxy = dxinv[0] * dxinv[1];
-    Real gamma = alpha * (dyy)*Real(2.0) + beta;
-    Real ccex =
-        - dyy * (ex(i  ,j-1,k  ) +
-                 ex(i  ,j+1,k  ))
-        + dxy * (ey(i  ,j-1,k  )
-               - ey(i  ,j  ,k  )
-               - ey(i+1,j-1,k  )
-               + ey(i+1,j  ,k  ));
-#else
-    Real dyy = dxinv[1] * dxinv[1];
-    Real dzz = dxinv[2] * dxinv[2];
-    Real dxy = dxinv[0] * dxinv[1];
-    Real dxz = dxinv[0] * dxinv[2];
-    Real gamma = alpha * (dyy+dzz)*Real(2.0) + beta;
-    Real ccex =
-        - dyy * (ex(i  ,j-1,k  ) +
-                 ex(i  ,j+1,k  ))
-        - dzz * (ex(i  ,j  ,k+1) +
-                 ex(i  ,j  ,k-1))
-        + dxy * (ey(i  ,j-1,k  )
-               - ey(i  ,j  ,k  )
-               - ey(i+1,j-1,k  )
-               + ey(i+1,j  ,k  ))
-        + dxz * (ez(i  ,j  ,k-1)
-               - ez(i  ,j  ,k  )
-               - ez(i+1,j  ,k-1)
-               + ez(i+1,j  ,k  ));
-#endif
-    Real res = rhs(i,j,k) - (gamma*ex(i,j,k) + alpha*ccex);
-    ex(i,j,k) += omega/gamma * res;
-}
 
-AMREX_GPU_DEVICE AMREX_FORCE_INLINE
-void mlcurlcurl_gsrb_y (int i, int j, int k,
-                        Array4<Real const> const& ex,
-                        Array4<Real      > const& ey,
-                        Array4<Real const> const& ez,
-                        Array4<Real const> const& rhs,
-                        Real alpha, Real beta,
-                        GpuArray<Real,AMREX_SPACEDIM> const& dxinv, int redblack)
-{
-    constexpr Real omega = Real(1.15);
+    Real dxx = adxinv[0] * adxinv[0];
+    Real dyy = adxinv[1] * adxinv[1];
 
-    if ((i+j+k+redblack) % 2 != 0) { return; }
+    if (((my_color == 0 || my_color == 3) && (color == 0 || color == 3)) ||
+        ((my_color == 1 || my_color == 2) && (color == 1 || color == 2)))
+    {
+        Real gamma = (dxx+dyy)*Real(2.0) + beta;
```

Comment:
Interestingly Github seems to highlight `gamma` as if it's a C++ keyword like `constexpr` or `if`. It's not, and I don't think it's a function defined in any standard header, either.